crates/bevy_ecs/src/query/iter.rs

```diff
@@ -508,7 +524,7 @@ impl<'w, 's, D: QueryData, F: QueryFilter> QueryIter<'w, 's, D, F> {
     /// # schedule.add_systems((system_1, system_2, system_3));
     /// # schedule.run(&mut world);
     /// ```
-    pub fn sort<L: ReadOnlyQueryData + 'w>(
+    pub fn sort<L: ReadOnlyQueryData + SingleEntityQueryData + 'w>(
```

Comment:
Doc comments about why `SingleEntityQueryData` is required for these methods would be helpful for users. It's not immediately clear why this is required.